lib/cartopy/mpl/clip_path.py

```diff
@@ -113,25 +54,14 @@ def intersection_point(p0, p1, p2, p3):
 # Provide a clip_path function which clips the given path to the given Bbox.
 # There is inbuilt mpl functionality with v1.2.1 and beyond, but we provide
 # a shim here for older mpl versions.
-if hasattr(mpath.Path, 'clip_to_bbox'):
-    def clip_path(subject, clip_bbox):
-        """
-        Clip the given path to the given bounding box.
-
-        """
-        return subject.clip_to_bbox(clip_bbox)
-else:
-    def clip_path(subject, clip_bbox):
-        """
-        Clip the given path to the given bounding box.
-
-        """
-        # A shim on clip_path_python to support Bbox path clipping.
-
-        bbox_patch = bbox_to_path(clip_bbox)
-        bbox_center = ((clip_bbox.x0 + clip_bbox.x1) / 2,
-                       (clip_bbox.y0 + clip_bbox.y1) / 2)
-        return clip_path_python(subject, bbox_patch, bbox_center)
+def clip_path(subject, clip_bbox):
+    """
+    Clip the given path to the given bounding box.
+
+    """
+    warnings.warn("This method has been deprecated. "
+                  "See the \"What's new\" section for v0.17.")
```

Comment:
I'd prefer the warning to contain the core information from the what's new so cross referencing is not necessary:

```
You can replace ``clip_path(subject, clip_bbox)`` by ``subject.clip_to_bbox(clip_bbox)``)
```

Reply:
Done (though for some reason GitHub's UI still shows the old warning.)